metadata-ingestion/src/datahub/ingestion/source/powerbi/m_query/resolver.py

```diff
@@ -0,0 +1,575 @@
+            ),
+            char='"',
+        )
+
+        if len(arguments) == 2:
+            # It is regular case of MS-SQL
+            LOGGER.debug("Handling with regular case")
+            return self.two_level_access_pattern(token_dict)
+
+        if len(arguments) >= 4 and arguments[2] != "Query":
+            LOGGER.debug("Unsupported case is found. Second index is not the Query")
+            return full_table_names
+
+        db_name: str = arguments[1]
+        tables: List[str] = native_sql_parser.get_tables(arguments[3])
+        for table in tables:
+            schema_and_table: List[str] = table.split(".")
+            if len(schema_and_table) == 1:
+                # schema name is not present. Default schema name in MS-SQL is dbo
+                # https://learn.microsoft.com/en-us/sql/relational-databases/security/authentication-access/ownership-and-user-schema-separation?view=sql-server-ver16
+                schema_and_table.insert(0, "dbo")
+
+            full_table_names.append(
+                f"{db_name}.{schema_and_table[0]}.{schema_and_table[1]}"
+            )
+        LOGGER.debug("MS-SQL full-table-names %s", full_table_names)
+
+        return full_table_names
+
+
+class OracleTableFullNameCreator(AbstractTableFullNameCreator):
+    def get_platform_pair(self) -> DataPlatformPair:
+        return SupportedDataPlatform.ORACLE.value
+
+    def _get_db_name(self, value: str) -> Optional[str]:
+        error_message: str = f"The target argument ({value}) should in the format of <host-name>:<port>/<db-name>[.<domain>]"
+        splitter_result: List[str] = value.split("/")
+        if len(splitter_result) != 2:
+            LOGGER.debug(error_message)
+            return None
+
+        db_name = splitter_result[1].split(".")[0]
+
+        return db_name
+
+    def get_full_table_names(self, token_dict: Dict[str, Any]) -> List[str]:
+        full_table_names: List[str] = []
+
+        LOGGER.debug("Processing Oracle token-dict %s", token_dict)
+
+        for data_access_function in token_dict:
+            arguments: List[str] = tree_function.remove_whitespaces_from_list(
+                tree_function.token_values(token_dict[data_access_function]["arg_list"])
+            )
+            # delete arg_list as we consumed it and don't want to process it in next step
+            del token_dict[data_access_function]["arg_list"]
+
+            for source in token_dict[data_access_function]:
+                source_dict: Dict[str, Any] = token_dict[data_access_function][source]
+
+                db_name: Optional[str] = self._get_db_name(arguments[0])
+                if db_name is None:
+                    return full_table_names
+
+                for schema in source_dict["item_selectors"]:
+                    schema_name: str = schema["items"]["Schema"]
+                    for item_selectors in source_dict[schema["assigned_to"]]:
+                        for item_selector in source_dict[schema["assigned_to"]][
```

Comment:
3 nested for loops - any way to reduce the nesting here? it's very hard to follow nesting in python.

Reply:
I resolved it, now we don't have loops